test/common/validation.go

```diff
@@ -92,14 +94,39 @@ func (client *Client) FindAnyLogContents(podName string, contents []string) (boo
 	if err != nil {
 		return false, err
 	}
+	eventContentsSet, err := parseEventContentsFromPodLogs(string(logs))
+	if err != nil {
+		return false, err
+	}
 	for _, content := range contents {
-		if strings.Contains(string(logs), content) {
+		if _, ok := eventContentsSet[content]; ok {
 			return true, nil
 		}
 	}
 	return false, nil
 }
 
+//Example log entry: 2019/08/21 22:46:38 {"msg":"Body-type1-source1--extname1-extval1-extname2-extvalue2","sequence":"1"}
+//Use regex to get the event content with json format: {"msg":"Body-type1-source1--extname1-extval1-extname2-extvalue2","sequence":"1"}
+//Get the eventContent with key "msg"
+//Returns a set(map implementation) with all unique event contents
+func parseEventContentsFromPodLogs(logs string) (map[string]bool, error) {
+	re := regexp.MustCompile(`{.+?}`)
```

Comment:
What does this regex match?

Reply:
It matches any substring starting with"{" and ending with "}".
For example, we have log string like 
" 2019/08/21 22:46:38 {"msg":"Body-type1-source1--extname1-extval1-extname2-extvalue2","sequence":"1"}  2019/08/21 23:15:38 {"msg":"Body-type1-source1--extname1-extval1","sequence":"1"} 2019/08/21 23:49:20 {"msg":"Body-type1-source1--extname3-extval13","sequence":"1"}

Reply:
For example, we have logs like this:
2019/08/21 21:46:38 {"msg":"Body-type1-source1--extname1-extval1-extname2-extvalue2","sequence":"1"} 2019/08/21 22:15:32 {"msg":"Body-type1-source1--extname1-extval1","sequence":"1"} 2019/08/21 23:32:12 {"msg":"Body-type1-source1--extname3-extvalue3","sequence":"1"}
With regex `{.+?}`, we will get these three strings:
{"msg":"Body-type1-source1--extname1-extval1-extname2-extvalue2","sequence":"1"}
{"msg":"Body-type1-source1--extname1-extval1","sequence":"1"} 
{"msg":"Body-type1-source1--extname3-extvalue3","sequence":"1"}
FYI: https://stackoverflow.com/questions/14213848/difference-between-and

Reply:
Ahh, I was trying to interpret the `{}` as special characters, not characters to match.

I don't think this will work if there is a nested object in the line. For example:
`{"msg":"foo", "bar":{"baz":"qux"}, "garply":"bam"}`. I think it will match `{"msg":"foo", "bar":{"baz":"qux"}`, leaving out the rest of the line. 

If we are certain the line will be JSON, we could do something like find the first `{`, then substring from there to the end of the line.